project/common.scala

```diff
@@ -48,33 +49,54 @@ object BuildCommon extends AutoPlugin {
     cancelable in Global := true,
 
     scalacOptions ++= Seq(
-      "-deprecation", "-feature", "-unchecked", "-encoding", "utf8"),
-    scalacOptions ++= Seq(
-      "-Ywarn-unused-import"),
-    scalacOptions ++= Seq(
+      "-deprecation",
+      "-encoding", "UTF-8",
+      "-feature",
+      "-language:existentials",
+      "-language:higherKinds",
       "-language:implicitConversions",
-      "-language:higherKinds"),
-
-    javacOptions ++= Seq("-encoding", "UTF-8", "-Xlint:-options"),
-    headers <<= (name, version) { (name, version) ⇒
-      Map(
-        "scala" → (
-          HeaderPattern.cStyleBlockComment,
-          s"""|/* -
-           | * Case Classy [$name]
-           | */
-           |
-           |""".stripMargin)
-      )
-    }
+      "-language:experimental.macros",
+      "-unchecked",
+      "-Xfatal-warnings",
+      "-Xlint",
+      "-Yno-adapted-args",
+      "-Ywarn-dead-code",
+      "-Ywarn-numeric-widen",
+      "-Ywarn-value-discard",
+      "-Ywarn-unused-import",
+      "-Xfuture",
+      "-Yno-predef",
+      "-Ypartial-unification"),
+
+    scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => Nil
+      case Some((2, 12)) => Seq("-Yliteral-types")
+      case _             => Nil
+    }),
+
+    libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => Seq(
+        compilerPlugin("com.milessabin" % "si2712fix-plugin" % "1.2.0" cross CrossVersion.full))
```

Comment:
I couldn't get 2.11 working w/ the typelevel compiler-- so I just did this instead. 2.11.9 should come out soon and then I can just use LB Scala.